lmn/models.py

```diff
@@ -58,3 +59,45 @@ class Note(models.Model):
     def __str__(self):
         return f'User: {self.user} Show: {self.show} Note title: {self.title} Text: {self.text} Posted on: {self.posted_date} Image: {self.image}'
 
+
+class Badge(models.Model):
+    name = models.CharField(max_length=50, blank=False)
+    description = models.CharField(max_length=200, blank=False)
+
+    def __str__(self):
+        return f'Name: {self.name}, Description: {self.description}'
+
+
+class Profile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    profile_image = models.ImageField(upload_to='user_profile_images/', blank=True, null=True)
+    shows_seen = models.ManyToManyField(Show)
+    bio = models.TextField(blank=True, null=True)
+    badges = models.ManyToManyField(Badge)
+
+
+    def save(self, *args, **kwargs):
+        old_profile = Profile.objects.filter(pk=self.pk).first()
+        if old_profile and old_profile.profile_image:
+            if old_profile.profile_image != self.profile_image:
+                self.delete_image(old_profile.profile_image)
+
+        super().save(*args, **kwargs)
+
+
+    def delete_image(self, image):
```

Comment:
Maybe Add a confirm message before deletion?

Reply:
I will add a confirm delete message later on. These methods ensure that the images themselves are deleted from the media directory when a profile image is changed or deleted. The user doesn't have the ability to edit their profile yet, but when that feature is implemented, we'll definitely want a message to confirm delete.  Thanks for the reminder!